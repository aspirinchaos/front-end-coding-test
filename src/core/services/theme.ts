import { DefaultTheme, IBreakpoint } from 'styled-components';

import { min, max } from './styled-responsive';

const BREAKPOINTS: IBreakpoint = {
  xs: 576,
  sm: 600,
  md: 992,
  lg: 1200,
  xl: 1700,
};

const theme: DefaultTheme = {
  breakpoints: BREAKPOINTS,
  colors: {
    main: '#e85a4f',
    secondary: '#8e8d8a',
    body: '#eae7dc',
  },
  font: 'Poppins,sans-serif',
  min,
  max,
};

export default theme;
