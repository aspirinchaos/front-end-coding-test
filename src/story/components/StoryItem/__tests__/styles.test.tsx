import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import theme from 'core/services/theme';

import {
  StoryContainer,
  Image,
  ImageSide,
  TextSide,
  Title,
  Info,
} from '../styles';

describe('StoryItem styles', () => {
  describe('StoryContainer', () => {
    it('renders correctly', () => {
      const { container } = render(
        <StoryContainer theme={theme}>Content</StoryContainer>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('has correct styles', () => {
      const { container } = render(
        <StoryContainer theme={theme}>Content</StoryContainer>,
      );
      expect(container.firstChild).toHaveStyleRule('display', 'flex');
      expect(container.firstChild).toHaveStyleRule('width', '50%');
      expect(container.firstChild).toHaveStyleRule('width', '100%', {
        media: `(max-width:${theme.breakpoints.md}px)`,
      });
      expect(container.firstChild).toHaveStyleRule('flex-wrap', 'wrap', {
        media: `(max-width:${theme.breakpoints.sm}px)`,
      });
      expect(container.firstChild).toHaveStyleRule('width', '33%', {
        media: `(min-width:${theme.breakpoints.xl}px)`,
      });
    });
  });
  describe('Image', () => {
    it('renders correctly', () => {
      const { container } = render(<Image />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('has correct styles', () => {
      const { container } = render(<Image />);
      expect(container.firstChild).toHaveStyleRule('height', 'auto');
      expect(container.firstChild).toHaveStyleRule('max-width', '100%');
    });
  });
  describe('ImageSide', () => {
    it('renders correctly', () => {
      const { container } = render(
        <ImageSide theme={theme}>Content</ImageSide>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('has correct styles', () => {
      const { container } = render(
        <ImageSide theme={theme}>Content</ImageSide>,
      );
      expect(container.firstChild).toHaveStyleRule('padding', '5px');
      expect(container.firstChild).toHaveStyleRule('width', '50%');
      expect(container.firstChild).toHaveStyleRule('width', '100%', {
        media: `(max-width:${theme.breakpoints.sm}px)`,
      });
    });
  });
  describe('TextSide', () => {
    it('renders correctly', () => {
      const { container } = render(<TextSide theme={theme}>Content</TextSide>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('has correct styles', () => {
      const { container } = render(<TextSide theme={theme}>Content</TextSide>);
      expect(container.firstChild).toHaveStyleRule('padding', '15px');
      expect(container.firstChild).toHaveStyleRule('width', '50%');
      expect(container.firstChild).toHaveStyleRule('width', '100%', {
        media: `(max-width:${theme.breakpoints.sm}px)`,
      });
    });
  });
  describe('Title', () => {
    it('renders correctly', () => {
      const { container } = render(<Title theme={theme}>Content</Title>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('has correct styles', () => {
      const { container } = render(<Title theme={theme}>Content</Title>);
      expect(container.firstChild).toHaveStyleRule('color', theme.colors.main);
      expect(container.firstChild).toHaveStyleRule('font-family', theme.font);
      expect(container.firstChild).toHaveStyleRule('font-size', '1.125rem');
      expect(container.firstChild).toHaveStyleRule('text-decoration', 'none');
      expect(container.firstChild).toHaveStyleRule('font-weight', '700');
      expect(container.firstChild).toHaveStyleRule('display', 'block');
      expect(container.firstChild).toHaveStyleRule('margin-bottom', '10px');
      expect(container.firstChild).toHaveStyleRule(
        'text-decoration',
        'underline',
        {
          modifier: ':hover',
        },
      );
    });
  });
  describe('Info', () => {
    it('renders correctly', () => {
      const { container } = render(<Info theme={theme}>Content</Info>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('has correct styles', () => {
      const { container } = render(<Info theme={theme}>Content</Info>);
      expect(container.firstChild).toHaveStyleRule(
        'color',
        theme.colors.secondary,
      );
      expect(container.firstChild).toHaveStyleRule('font-family', theme.font);
      expect(container.firstChild).toHaveStyleRule('font-size', '0.875rem');
      expect(container.firstChild).toHaveStyleRule('font-weight', '300');
      expect(container.firstChild).toHaveStyleRule('margin-bottom', '5px');
    });
  });
});
