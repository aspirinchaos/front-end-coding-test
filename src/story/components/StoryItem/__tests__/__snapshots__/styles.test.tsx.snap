// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryItem styles Image renders correctly 1`] = `
.c0 {
  max-width: 100%;
  height: auto;
}

<img
  class="c0"
/>
`;

exports[`StoryItem styles ImageSide renders correctly 1`] = `
.c0 {
  padding: 5px;
  width: 50%;
}

@media (max-width:600px) {
  .c0 {
    width: 100%;
  }
}

<div
  class="c0"
>
  Content
</div>
`;

exports[`StoryItem styles Info renders correctly 1`] = `
.c0 {
  color: #8e8d8a;
  font-family: Poppins,sans-serif;
  font-size: 0.875rem;
  font-weight: 300;
  margin-bottom: 5px;
}

<div
  class="c0"
>
  Content
</div>
`;

exports[`StoryItem styles StoryContainer renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50%;
}

@media (max-width:992px) {
  .c0 {
    width: 100%;
  }
}

@media (min-width:1700px) {
  .c0 {
    width: 33%;
  }
}

@media (max-width:600px) {
  .c0 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

<div
  class="c0"
>
  Content
</div>
`;

exports[`StoryItem styles TextSide renders correctly 1`] = `
.c0 {
  padding: 15px;
  width: 50%;
}

@media (max-width:600px) {
  .c0 {
    width: 100%;
  }
}

<div
  class="c0"
>
  Content
</div>
`;

exports[`StoryItem styles Title renders correctly 1`] = `
.c0 {
  color: #e85a4f;
  font-size: 1.125rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Poppins,sans-serif;
  font-weight: 700;
  display: block;
  margin-bottom: 10px;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  class="c0"
>
  Content
</a>
`;
